_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+      - run:
+          name: Give Executable Power
+          command: 'chmod +x gradlew'
+      - run:
+          name: Backend tests
+          command: npm run ci:backend:test
+      - run:
+          name: Build Spring Boot Docker Image
+          command: ./gradlew bootBuildImage --imageName=$IMAGE_NAME
+      - run:
+          name: Publish Docker Image
+          command: |
+            docker login -u your-dockerhub-username -p $DOCKERHUB_PASS
+            docker push $IMAGE_NAME
+```
+
+Do the same updates to the `gateway/.circleci/config.yml` file. Be sure to replace `your-dockerhub-username` and set the correct `IMAGE_NAME`.
+
+To take advantage of the one-step GitHub integration of CircleCI, you need a [GitHub](https://github.com/signup) account. After signing in to GitHub, create a new public repository `store`. Follow the instructions to push the existing repository from your local using the command line. Do the same with the `gateway` project.
+
+[Sign up](https://circleci.com/integrations/github) at CircleCI with your GitHub account, and on the left menu choose **Projects**. Find the `store` project and click **Set Up Project**. Configure the project to **Use the `.circleci/config.yml` in my repo**.
+
+{% img blog/jhipster-ci-cd/circleci-project.png alt:"CircleCI project setup form" width:"450" %}{: .center-image }
+
+Do the same for the `gateway` project. The configuration triggers an initial pipeline execution that might fail, because you still must set up DockerHub credentials for both projects, allowing CircleCI to push the container images. At the `store` project page, on the top right, choose **Project Settings**. Then choose **Environment Variables**. Click **Add Environment Variable**. and set the following values:
+
+- Name: DOCKERHUB_PASS
+- Value: your DockerHub password, or better, a DockerHub access token if you have 2FA enabled.
+
+**Note**: For creating a DockerHub access token, sign in to DockerHub, and choose **Account Settings** in the top right user menu. Then, on the left menu, choose **Security**. Click **New Access Token** and set a description for the token, for example, _circleci-store_. Then click **Generate** and copy the new token.
+
+Once a project is set up in CircleCI, a pipeline is triggered each time a commit is pushed to the configured branch. The pipeline in execution appears on the **Dashboard** page. You can also manually trigger the pipeline from the **Dashboard** if you choose the project and branch from the pipeline filters, and then click **Trigger Pipeline**. Before moving on to the next section, manually execute the store pipeline and the gateway pipeline once, to push the first image of each to DockerHub.
+
+{% img blog/jhipster-ci-cd/circleci-job-success.png alt:"CircleCI job success" width:"800" %}{: .center-image }
+
+## Install Spinnaker on Google Kubernetes Engine
+
+Spinnaker supports the following installation environments:
```

Comment:
It seems like we should have a paragraph that explains what Spinnaker is here and how it differs from CircleCI. When I first heard of it, I thought it was an all-in-one solution that did CI and CD.

Reply:
Added some comments.